dvc/repo/ls.py

```diff
@@ -0,0 +1,99 @@
+import os
+
+
+@staticmethod
+def ls(url, targets=[], rev=None, recursive=None):
+    if os.path.exists(url):
+        return _ls_local_repo(url, targets, recursive)
+
+    return _ls_git_repo(url, targets, rev, recursive)
+
+
+def _ls_files_local_repo(path, targets=[], recursive=None):
+    from dvc.repo import Repo
+
+    repo = Repo(path)
+    files = []
+    if not len(targets):
+        targets.append(".")
+
+    def prettify(prefix, filename, is_root):
+        if not is_root:
+            filename = os.path.join(prefix, filename)
+
+        if os.path.isdir(filename):
+            return f"{filename}/"
+        return filename
+
+    for target in targets:
+        target_path = os.path.join(path, target)
+        if not os.path.exists(target_path):
+            continue
+
+        for dirpath, dirnames, filenames in repo.tree.walk(target_path):
+            prefix = os.path.relpath(dirpath, path)
+            is_root = os.path.abspath(dirpath) == os.path.abspath(path)
+            files.extend(map(lambda f: prettify(prefix, f, is_root), dirnames))
+            files.extend(
+                map(lambda f: prettify(prefix, f, is_root), filenames)
+            )
+            if not recursive:
+                dirnames.clear()
```

Comment:
No need to modify `dirnames`
```suggestion
                break
```